src/tms_service.ts

```diff
@@ -110,6 +132,94 @@ export class TMSService extends AbstractEmsService {
   }
 
+  /*
+  This method returns an array of objects with the layers and the new
+  layout['text-field'] to apply using map.setLayoutProperty
+  */
+  public static transformLanguageProperty(
+    layer: LayerSpecification,
+    lang: keyof typeof TMSService.SupportedLanguages
+  ): DataDrivenPropertyValueSpecification<FormattedSpecification> | undefined {
+    const newLayer = { ...layer };
+    const omtLang = this.SupportedLanguages[lang].omtCode;
+
+    if (newLayer.type === 'symbol' && newLayer.layout !== undefined) {
+      const textField = newLayer.layout['text-field'];
+      if (textField && typeof textField === 'string') {
+        return TMSService._getTextField(textField, omtLang);
+      }
+    }
+    return;
+  }
+
+  /*
+  This method returns an array of objects per layer, containing a list of
+  properties with new colors to apply using map.setPaintProperty
+  */
+  public static transformColorProperties(
+    layer: LayerSpecification,
+    color: string,
+    operation: blendMode,
+    percentage: number
+  ): { property: keyof layerPaintProperty; color: mbColorDefinition | undefined }[] {
+    if (['background', 'fill', 'line', 'symbol'].indexOf(layer.type) !== -1 && layer.paint) {
+      const paint = layer.paint as layerPaintProperty;
+      const types = Object.keys(paint).filter((key) => {
+        return (
+          [
+            'background-color',
+            'circle-color',
+            'circle-stroke-color',
+            'fill-color',
+            'fill-extrusion-color',
+            'fill-outline-color',
+            'icon-color',
+            'icon-halo-color',
+            'line-color',
+            'text-color',
+            'text-halo-color',
+          ].indexOf(key) !== -1
+        );
+      }) as Array<keyof layerPaintProperty>;
+      return types.map((type) => {
+        const paintColor = paint[type];
+        return {
+          property: type,
+          color: paintColor ? colorizeColor(paintColor, color, operation, percentage) : paintColor,
+        };
+      });
+    } else {
+      return [];
+    }
+  }
+
+  private static _getTextField(
+    label: string,
+    lang: string
+  ): DataDrivenPropertyValueSpecification<string> {
+    // Capture {name:xx} labels
+    const labelMatch = label.match(/\{name:(.{2})\}/);
+    if (labelMatch && labelMatch[1] != lang) {
+      // Only apply if the languages are different
+      return ['coalesce', ['get', `name:${lang}`], ['get', `name:${labelMatch[1]}`]];
+    } else {
+      if (label.includes('latin') && label.includes('nonlatin')) {
+        // Capture the common pattern {name:latin}\n{name:nonlatin}
+        return [
+          'coalesce',
+          ['get', `name:${lang}`],
+          ['concat', ['get', 'name:latin'], '\n', ['get', 'name:nonlatin']],
+        ];
+      } else if (label.includes('name')) {
+        // Capture any other label using `name`
+        return ['coalesce', ['get', `name:${lang}`], ['get', 'name:latin']];
+      }
```

Comment:
I may be misunderstanding something here. Why do we need to have this nested conditional?

Can we just return the concatenated latin and non-latin strings whenever the `name:${lang}` is null?
```suggestion
        return [
          'coalesce',
          ['get', `name:${lang}`],
          ['concat', ['get', 'name:latin'], '\n', ['get', 'name:nonlatin']],
        ];
```